PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/sparkserverless/common/ErrorMessageTooltip.java

```diff
@@ -0,0 +1,34 @@
+package com.microsoft.azure.sparkserverless.common;
+
+import com.intellij.ui.JBColor;
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import com.microsoft.azuretools.azurecommons.helpers.Nullable;
+
+import javax.swing.*;
+import java.awt.*;
+
+public class ErrorMessageTooltip extends JToolTip{
+    @Nullable
+    private Popup popup;
+
+    public ErrorMessageTooltip(@NotNull JComponent component) {
+        super();
+        setComponent(component);
+        setForeground(JBColor.RED);
+    }
+
+    void hideToolTip() {
+        if (popup != null) {
+            popup.hide();
+        }
+    }
+
+    void showErrorToolTip(@NotNull Point location, @Nullable String errorMsg) {
+        try {
+            setTipText(errorMsg);
+            popup = PopupFactory.getSharedInstance().getPopup(getComponent(), this, location.x, location.y - 20);
```

Comment:
Would it be conflicted with the component common tooltip?

Reply:
Confirmed, it doesn't conflict with tooltip of `JComponent`. Both tooltips can show as expected.

---
In reply to: [199710235](https://github.com/Microsoft/azure-tools-for-java/pull/1747#discussion_r199710235) [](ancestors = 199710235)